syntax = "proto3";

package transaction_verification;

message InitOrderRequest {
  string orderId = 1;
  string checkoutData = 2;
}

message InitOrderResponse {
  bool success = 1;
  string clock = 2;
}

message VerificationRequest {
  string orderId = 1;
  string clock = 2;
}

message VerificationResponse {
  bool success = 1;
  string message = 2;
  string clock = 3;
}

service VerificationService {
  rpc InitOrder(InitOrderRequest) returns (InitOrderResponse);
  rpc VerifyItems(VerificationRequest) returns (VerificationResponse);
  rpc VerifyUserData(VerificationRequest) returns (VerificationResponse);
  rpc VerifyCreditCard(VerificationRequest) returns (VerificationResponse);
}
